wemake_python_styleguide/logic/tree/assignments.py

```diff
@@ -0,0 +1,87 @@
+import ast
+import itertools
+from typing import Iterable, List, Optional, Tuple
+
+from wemake_python_styleguide import constants
+from wemake_python_styleguide.compat.aliases import AssignNodes
+from wemake_python_styleguide.compat.functions import get_assign_targets
+from wemake_python_styleguide.logic import nodes
+from wemake_python_styleguide.types import AnyAssign
+
+#: Type alias for the assignments we return from class inspection.
+_AllAssignments = Tuple[List[AnyAssign], List[AnyAssign]]
+
+
+def flat_assignment_values(assigns: Iterable[AnyAssign]) -> Iterable[ast.AST]:
+    """
+    Returns flat values from assignment.
+
+    Use this function when you need to get list of values
+    from assign nodes.
+    """
+    return itertools.chain.from_iterable((
+        _flat_nodes(assign.value)
+        for assign in assigns
+        if isinstance(assign.value, ast.AST)
+    ))
+
+
+def _flat_nodes(node: ast.AST) -> List[ast.AST]:
+    flatten_nodes: List[ast.AST] = []
+
+    if isinstance(node, ast.Tuple):
+        for subnode in node.elts:
+            flatten_nodes.extend(_flat_nodes(subnode))
+    else:
+        flatten_nodes.append(node)
+    return flatten_nodes
+
+
+def get_assignments(node: ast.ClassDef) -> _AllAssignments:
```

Comment:
This function is really similar to https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/logic/tree/classes.py#L76 

Can you please exaplain, why can't we reuse it?

Reply:
I wrote another one function for finding assignments only. 
[This one](https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/logic/tree/classes.py#L76) finds assignments for classes and attributes for instances and I decided to unify returned values for lambda assignment violations.

I agree that both functions are similar and I need to think about how to decreases code duplication.